### PR TITLE
iox-#2184 Enable clang-tidy for all of hoofs

### LINKED ISSUE #2184: Activate clang-tidy for all the code in iceoryx_hoofs
## Brief feature description

Currently there are quite a lot of entries in `clang-tidy-diff-scan.txt`. This was done to be able to gradually fix the clang tidy warning. All the production code in iceoryx_hoofs and most of the tests are now warning free and the `clang-tidy-diff-scan.txt` list can be simplified to `./iceoryx_hoofs/**/*`.

### PR DESCRIPTION
**Pull request by elBoberido**
## Pre-Review Checklist for the PR Author

1. [x] Add a second reviewer for complex new features or larger refactorings
1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] All touched (C/C++) source code files from `iceoryx_hoofs` are added to `./clang-tidy-diff-scans.txt`
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Notes for Reviewer

The PR adds all files of iceoryx_hoofs to the list of files to scan by clang-tidy and fixes the clang-tidy warnings on the new files.

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] All touched (C/C++) source code files from `iceoryx_hoofs` have been added to `./clang-tidy-diff-scans.txt`
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #2184

### CODE REVIEW

File: iceoryx_hoofs/test/stresstests/benchmark_optional_and_expected/benchmark_optional_and_expected.cpp [hunk 1]
```diff
@@ -28,21 +29,21 @@ template <uint64_t Size>
...
 
 template <uint64_t Size>
 struct LargeObjectComplexCTor
 {
     LargeObjectComplexCTor()
     {
-        for (uint64_t i = 0u; i < Size; ++i)
+        for (auto& element : dataBlob)
         {
-            dataBlob[i] = static_cast<char>((++globalCounter) % 256);
```
Comment (mossmaurice) [inline iceoryx_hoofs/test/stresstests/benchmark_optional_and_expected/benchmark_optional_and_expected.cpp:41]: I read `dataBob` :rofl:

**Review by mossmaurice — APPROVED** [review]
LGTM, thanks for the PR!

**Comment by mossmaurice** [comment]
@elBoberido Up to you on whether you want to add your copyright.

**Comment by elBoberido** [comment]
> @elBoberido Up to you on whether you want to add your copyright.

Na, not for clang-tidy fixes :)

**Pull request merged**